docs/architecture/adr-056-proving-amnesia-attack.md

```diff
@@ -0,0 +1,106 @@
+
+## Changelog
+
+- 02.04.20: Initial Draft
+- 06.04.20: Second Draft
+
+## Context
+
+Whilst most created evidence of malicious behaviour is self evident such that any individual can verify them independently there are types of evidence, known as global evidence, that require further collaboration from the network in order to accumulate enough information to create evidence that is individually verifiable and can therefore be processed through consensus. Fork Accountability as a whole constitutes of detection, proving and punishing. This ADR addresses how to prove infringement from global evidence that is sent to a full node.
+
+The currently only known form of global evidence stems from [flip flopping](https://github.com/tendermint/spec/blob/master/spec/consensus/light-client/accountability.md#flip-flopping) attacks. The schematic below explains one scenario where an amnesia attack, a form of flip flopping, can occur such that C1 and C2 commit different blocks.
+
+![](../imgs/tm-amnesia-attack.png)
+
+This creates a fork on the main chain.  Back to the past, another form of flip flopping, creates a light fork (capable of fooling those not involved in consensus), in a similar way, with F taking the precommits from C1 and forging a commit from them.
+
+## Decision
+
+As the distinction between these two attacks (amnesia and back to the past) can only be distinguished by confirming with all validators (to see if it is a full fork or a light fork), for the purpose of simplicity, these attacks will be treated as the same.
+
+Currently, the evidence reactor is used to simply broadcast and store evidence. Instead of perhaps creating a new reactor for the specific task of verifying these attacks, the current evidence reactor will be extended.
+
+The process begins with a light client receiving conflicting headers (in the future this could also be a full node during fast sync), which it sends to a full node to analyse. As part of [evidence handling](https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-047-handling-evidence-from-light-client.md), this could be deduced into potential amnesia evidence
+
+```
+type PotentialAmnesiaEvidence struct {
+	V1 []*types.Vote
+	V2 []*types.Vote
+
+	timestamp time.Time
+}
+```
+
+*REMARK: We could also use `VoteSet`*
+
+*NOTE: this is almost identical to Duplicate Vote Evidence and although one is for different rounds and the other is for the same round, there is a case that these two could be combined. In addition, I believe batch processing to be more efficient than individual processing and would think all evidence that constitutes multiple attackers should be grouped together.*
+
+The evidence should contain the precommit votes for the intersection of validators that voted for both rounds. The votes should be all valid and the height and time that the infringement was made should be within:
+
+`Bonding period - Amnesia trial period - Gossip safety margin`
```

Comment:
It might be good to define these terms first. Also, AFAIK, Tendermint shouldn't make use of the nomenclature "Bonding Period", rather safety margin or something else.

Reply:
Why is that? Is bonding period more application specific?

Reply:
Yes, it is application-specific. I believe @melekes uses Trusting Period?

Reply:
Yes, trusting period is the correct term.

Reply:
Okay I will switch to trusting period